### PR TITLE
Option to not add _total suffix to counter metrics in Prometheus exporter

### PR DESCRIPTION
**Pull request by Bassardes**
Fixes #
Design discussion issue #

## Changes

According to this [doc](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/prometheus_and_openmetrics.md#sums)

> If the metric name for monotonic Sum metric points does not end in a suffix of _total a suffix of _total MUST be added by default, otherwise the name MUST remain unchanged. Exporters SHOULD provide a configuration option to disable the addition of _total suffixes. 

This PR adds option to disable addition of _total suffix for counter metrics in Prometheus exporter

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Prometheus.AspNetCore/PrometheusAspNetCoreOptions.cs [hunk 1]
```diff
@@ -17,6 +17,15 @@ public class PrometheusAspNetCoreOptions
     /// </summary>
     public string ScrapeEndpointPath { get; set; } = DefaultScrapeEndpointPath;
 
+    /// <summary>
+    /// Gets or sets a value indicating whether addition of _total suffix for counter metric names is disabled.
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Exporter.Prometheus.AspNetCore/PrometheusAspNetCoreOptions.cs:21]:
```suggestion
    /// Gets or sets a value indicating whether addition of _total suffix for counter metric names is disabled. Default value: <see langword="false"/>.
```

File: src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusExporterOptions.cs [hunk 2]
```diff
@@ -28,4 +28,9 @@ public int ScrapeResponseCacheDurationMilliseconds
             this.scrapeResponseCacheDurationMilliseconds = value;
         }
     }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether addition of _total suffix for counter metric names is disabled.
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusExporterOptions.cs:33]:
```suggestion
    /// Gets or sets a value indicating whether addition of _total suffix for counter metric names is disabled. Default value: <see langword="false"/>.
```

**Comment by Bassardes** [comment]
I don't understand what happens with tests, locally it's all passes. I need help with it

**Comment by vishweshbankwar** [comment]
@Bassardes Could you address this https://github.com/open-telemetry/opentelemetry-dotnet/pull/5305#issuecomment-1922143862

**Comment by Bassardes** [comment]
@vishweshbankwar Seems that I can't, because this merge [c0eb638](https://github.com/open-telemetry/opentelemetry-dotnet/commit/c0eb6384d08ad6ab2142103b7224b2e118b67f38) wasn't done by me. I will try force push and merge master by myself

File: src/OpenTelemetry.Exporter.Prometheus.AspNetCore/PrometheusAspNetCoreOptions.cs [hunk 3]
```diff
@@ -17,6 +17,15 @@ public class PrometheusAspNetCoreOptions
     /// </summary>
     public string ScrapeEndpointPath { get; set; } = DefaultScrapeEndpointPath;
 
+    /// <summary>
+    /// Gets or sets a value indicating whether addition of _total suffix for counter metric names is disabled. Default value: <see langword="false"/>.
+    /// </summary>
+    public bool DisableTotalNameSuffixForCounters
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.Prometheus.AspNetCore/PrometheusAspNetCoreOptions.cs:23]: `public bool EnableCounterTotalSuffix = true` - alternate suggestion.

The default should be true as per https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/prometheus_and_openmetrics.md#sums

Comment (Bassardes) [reply src/OpenTelemetry.Exporter.Prometheus.AspNetCore/PrometheusAspNetCoreOptions.cs:23]: > Exporters SHOULD provide a configuration option to disable..

So I think _DisableTotalNameSuffixForCounters_ will be better

Comment (utpilla) [inline src/OpenTelemetry.Exporter.Prometheus.AspNetCore/PrometheusAspNetCoreOptions.cs:23]: Add this for [PrometheusHttpListenerOptions](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.Prometheus.HttpListener/PrometheusHttpListenerOptions.cs) as well.

Comment (utpilla) [reply src/OpenTelemetry.Exporter.Prometheus.AspNetCore/PrometheusAspNetCoreOptions.cs:23]: Add a CHANGELOG entry as well.

File: test/OpenTelemetry.Exporter.Prometheus.HttpListener.Tests/PrometheusMetricTests.cs [hunk 4]
```diff
@@ -118,84 +118,91 @@ public void Unit_AnnotationMismatch_Close()
...
     {
-        AssertName("metric_name", "2", PrometheusType.Summary, "metric_name_2");
+        AssertName("metric_name", "2", PrometheusType.Summary, false, "metric_name_2");
     }
 
     [Fact]
     public void Name_UnitAlreadyPresentInName_NotAppended()
     {
-        AssertName("metric_name_total", "total", PrometheusType.Counter, "metric_name_total");
+        AssertName("metric_name_total", "total", PrometheusType.Counter, false, "metric_name_total");
```
[the remark this answers is not in the document]
Comment (Bassardes) [reply test/OpenTelemetry.Exporter.Prometheus.HttpListener.Tests/PrometheusMetricTests.cs:175]: done

File: src/OpenTelemetry.Exporter.Prometheus.AspNetCore/PrometheusAspNetCoreOptions.cs (hunk 3 above)
Comment (Bassardes) [reply src/OpenTelemetry.Exporter.Prometheus.AspNetCore/PrometheusAspNetCoreOptions.cs:23]: done

File: src/OpenTelemetry.Exporter.Prometheus.AspNetCore/CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+* Added option to disable _total suffix addition to counter metrics in
```
Comment (utpilla) [inline src/OpenTelemetry.Exporter.Prometheus.AspNetCore/CHANGELOG.md:5]: No need to mention "in Prometheus exporter". This CHANGELOG file is specific to Prometheus exporter.

File: src/OpenTelemetry.Exporter.Prometheus.HttpListener/CHANGELOG.md [hunk 6]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+* Added option to disable _total suffix addition to counter metrics in
```
Comment (utpilla) [inline src/OpenTelemetry.Exporter.Prometheus.HttpListener/CHANGELOG.md:5]: Same as https://github.com/open-telemetry/opentelemetry-dotnet/pull/5305/files#r1493103785

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry.Exporter.Prometheus.AspNetCore/CHANGELOG.md (hunk 5 above)
Comment (Bassardes) [reply src/OpenTelemetry.Exporter.Prometheus.AspNetCore/CHANGELOG.md]: fixed

File: src/OpenTelemetry.Exporter.Prometheus.HttpListener/CHANGELOG.md (hunk 6 above)
Comment (Bassardes) [reply src/OpenTelemetry.Exporter.Prometheus.HttpListener/CHANGELOG.md]: fixed

**Review by CodeBlanch — APPROVED** [review]
LGTM

**Review by rajkumar-rangaraj — APPROVED** [review]

**Pull request merged**